examples/audio/transformer_asr.py

```diff
@@ -0,0 +1,523 @@
+    def __init__(
+        self, input_type="tokens", num_vocab=1000, num_hid=64, num_ff=128, maxlen=100,
+    ):
+        super().__init__()
+        self.input_type = input_type
+        if input_type == "tokens":
+            self.emb = tf.keras.layers.Embedding(num_vocab, num_hid)
+        elif input_type == "features":
+            self.conv1 = tf.keras.layers.Conv1D(
+                num_hid, 11, strides=2, padding="same", activation="relu"
+            )
+            self.conv2 = tf.keras.layers.Conv1D(
+                num_hid, 11, strides=2, padding="same", activation="relu"
+            )
+            self.conv3 = tf.keras.layers.Conv1D(
+                num_hid, 11, strides=2, padding="same", activation="relu"
+            )
+        else:
+            raise ValueError("input_type must be one of ('tokens', 'features')")
+        self.pos_emb = layers.Embedding(input_dim=maxlen, output_dim=num_hid)
+
+    def call(self, x):
+        if self.input_type == "tokens":
+            maxlen = tf.shape(x)[-1]
+            x = self.emb(x)
+            positions = tf.range(start=0, limit=maxlen, delta=1)
+            positions = self.pos_emb(positions)
+            return x + positions
+        else:
+            x = self.conv1(x)
+            x = self.conv2(x)
+            return self.conv3(x)
+
+
+"""
+## Transformer Encoder Layer
+"""
+
+
+class TransformerEncoderLayer(layers.Layer):
+    def __init__(self, embed_dim, num_heads, ff_dim, rate=0.1):
+        super().__init__()
+        self.att = layers.MultiHeadAttention(num_heads=num_heads, key_dim=embed_dim)
+        self.ffn = keras.Sequential(
+            [layers.Dense(ff_dim, activation="relu"), layers.Dense(embed_dim),]
+        )
+        self.layernorm1 = layers.LayerNormalization(epsilon=1e-6)
+        self.layernorm2 = layers.LayerNormalization(epsilon=1e-6)
+        self.dropout1 = layers.Dropout(rate)
+        self.dropout2 = layers.Dropout(rate)
+
+    def call(self, inputs, training):
+        attn_output = self.att(inputs, inputs)
+        attn_output = self.dropout1(attn_output, training=training)
+        out1 = self.layernorm1(inputs + attn_output)
+        ffn_output = self.ffn(out1)
+        ffn_output = self.dropout2(ffn_output, training=training)
+        return self.layernorm2(out1 + ffn_output)
+
+
+"""
+## Transformer Decoder Layer
+"""
+
+
+class TransformerDecoderLayer(layers.Layer):
+    def __init__(self, embed_dim, num_heads, ff_dim, rate=0.1):
+        super().__init__()
+        self.ln1 = layers.LayerNormalization(epsilon=1e-6)
+        self.ln2 = layers.LayerNormalization(epsilon=1e-6)
+        self.ln3 = layers.LayerNormalization(epsilon=1e-6)
+        self.self_att = layers.MultiHeadAttention(
+            num_heads=num_heads, key_dim=embed_dim
+        )
+        self.enc_att = layers.MultiHeadAttention(num_heads=num_heads, key_dim=embed_dim)
+        self.self_drop = layers.Dropout(0.5)
+        self.enc_drop = layers.Dropout(0.0)
```

Comment:
A dropout layer with no dropout does nothing, you can remove it, or otherwise set the rate to a proper value